docs/gallery/howto/autogen/aggregate.py

```diff
@@ -0,0 +1,329 @@
+
+# we have to increase the link limit because by default workgraph only supports
+# one link per input socket this is still an experimental feature that is why
+aggregate_task.inputs["collected_ints"].link_limit = 50
+aggregate_task.inputs["collected_floats"].link_limit = 50
+
+
+for i in range(2):  # this can be chosen as wanted
+    seed = Int(i)
+    generator_int_task = wg.add_task(generator_int, name=f"generator_int{i}", seed=seed)
+    generator_float_task = wg.add_task(
+        generator_float, name=f"generator_float{i}", seed=seed
+    )
+
+    wg.add_link(
+        generator_int_task.outputs["result"],
+        aggregate_task.inputs["collected_ints"],
+    )
+
+    wg.add_link(
+        generator_float_task.outputs["result"],
+        aggregate_task.inputs["collected_floats"],
+    )
+wg.to_html()
+# %%
+# Run the workgraph
+
+wg.run()
+
+# %%
+# Print the output
+print("aggregate_task int_sum", aggregate_task.outputs["int_sum"].value)
+print("aggregate_task float_sum", aggregate_task.outputs["float_sum"].value)
+
+
+# %%
+# Plot provenance
+
+generate_node_graph(wg.pk)
+
+# %%
+# Using context for dynamic outputs
+# =================================
+# If are not familiar with context please refer to the doc page `Use Context to pass data between tasks`.
+
+
+@task.calcfunction()
+def generator(seed: Int):
+    import random
+
+    random.seed(seed.value)
+    return Int(random.randint(0, 1))  # one can also use
+
+
+@task.calcfunction()
+def aggregate(**collected_values):  # We use a keyword argument to obtain a dictionary
+    for key, value in collected_values.items():
+        print("key:", key, "value:", value)
+    return {"result": sum(collected_values.values())}
+
+
+# For this use case it is more convenient to use the graph_builder as we can
+# expose the context under a more convenient name.
+@task.graph_builder(
+    outputs=[{"name": "result", "from": "context.generated"}]
+)  # this port is created by `set_context`
+def generator_loop(nb_iterations: Int):
+    wg = WorkGraph()
+    for i in range(nb_iterations.value):  # this can be chosen as wanted
+        generator_task = wg.add_task(generator, name=f"generator{i}", seed=Int(i))
+        generator_task.set_context({"result": f"generated.seed{i}"})
+    return wg
+
+
+wg = WorkGraph("generate_aggregate_by_context")
+
+
+generator_loop_task = wg.add_task(
+    generator_loop, name="generator_loop", nb_iterations=Int(2)
+)
+
+aggregate_task = wg.add_task(
+    aggregate,
+    name="aggregate_task",
+    collected_values=generator_loop_task.outputs["result"],
+)
+
+wg.to_html()
+
+
+# %%
+# Run the workgrhap
+
+wg.run()
+
+# %%
+# Print the output
+
+print("aggregate_task result", aggregate_task.outputs["result"].value)
+
+
+# %%
+# Plot provenance
+
+generate_node_graph(wg.pk)
+
+# %%
+# To support multiple dynamically sized inputs we can add another context and link it.
+
+
+# %%
+# Nested dynamic inputs for a calcfunction
+# ========================================
```

Comment:
I suggest removing this section. Then, add a short paragraph at the beginning of the notebook to explain why we need to use the keyword argument. This is important because all the calcfunctions used in the sections are using it.

Reply:
Hm... Maybe we can move this to the [port section](https://aiida-workgraph.readthedocs.io/en/latest/concept/autogen/socket.html)?

Reply:
OK

Reply:
I feel like how it was written does not fit so well into the port section, but I saved it outside of this PR for some other time. I redid the introduction